File: scripts/parity/scenarios/js/midpoint_polar.mjs
// Parity scenario: midpoint and polar_to_point geometry functions
import { Point, polar_to_point, midpoint, transform } from '../../../../dist/index.js'

// Test midpoint and polar_to_point geometry functions
const pt1 = new Point({ x: 40, y: 40, z: 0.2 })
const pt2 = new Point({ x: 40, y: 60, z: 0.2 })
const pt3 = polar_to_point(pt2, 15, Math.PI / 4)  // tau/8 = 2π/8 = π/4
const pt4 = midpoint(pt1, pt2)

const steps = [pt1, pt2, pt3, pt4]

const g = transform(steps, 'gcode', { show_tips: false, show_banner: false }).gcode
process.stdout.write(g.trimEnd() + '\n')

File: scripts/parity/scenarios/py/midpoint_polar.py
import fullcontrol as fc

# Test midpoint and polar_to_point geometry functions
from math import tau

pt1 = fc.Point(x=40, y=40, z=0.2)
pt2 = fc.Point(x=40, y=60, z=0.2)
pt3 = fc.polar_to_point(pt2, 15, tau/8)
pt4 = fc.midpoint(pt1, pt2)

steps = [pt1, pt2, pt3, pt4]

res = fc.transform(steps, result_type='gcode')
print(res)
